sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.resolved) {
```

Comment:
Unfortunately, to use `val nullable = encoder.clsTag.runtimeClass.isPrimitive` causes `NullPointerException` since `Class[java.lang.Long].isPrimitive` returns `true`.
We want to get `false` for `Class[java.lang.Long]` and get `true` for `(LongType, false)`.